heron/tmaster/src/cpp/manager/tcontroller.cpp

```diff
@@ -262,6 +268,44 @@ void TController::HandleUpdateRuntimeConfigRequestDone(IncomingHTTPRequest* requ
   delete request;
 }
 
+void TController::HandleGetCurPPlanRequest(IncomingHTTPRequest* request) {
+  LOG(INFO) << "Got a GetCurPPlan request from " << request->GetRemoteHost() << ":"
+              << request->GetRemotePort();
+
+  // make sure all the stream managers are alive, in case that when container is fail,
+  // physical plan is still available at TMaster but not a valid one.
+  if (tmaster_->GetStmgrsRegSummary()->absent_stmgrs_size() != 0) {
+      http_server_->SendErrorReply(request, 400);
+      delete request;
+      return;
+  }
+
+  if (tmaster_->getPhysicalPlan() == NULL) {
+    http_server_->SendErrorReply(request, 400);
+    delete request;
+    return;
+  } else {
+    std::string pplanString;
+    tmaster_->getPhysicalPlan()->SerializeToString(&pplanString);
+    std::string pplanStringFixed = "";
+
+    // convert original pplan string to a new one which do not contains b'00000000
```

Comment:
suggest changing the comment to:

SerializeToString() returns object in binary format which needs to be encoded.